src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -87,60 +90,51 @@ public void addBinaryField(FieldInfo field, DocValuesProducer valuesProducer) th
     public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
         KNNCounter.GRAPH_INDEX_REQUESTS.increment();
         if (field.attributes().containsKey(KNNVectorFieldMapper.KNN_FIELD)) {
-
-            // Get all parameters from field attributes
-            Map<String, String> fieldAttributes = field.attributes();
-            Map<String, Object> parameters = new HashMap<>();
-
-            String engineName = fieldAttributes.getOrDefault(KNNConstants.KNN_ENGINE, KNNEngine.DEFAULT.getName());
+            // Get engine to be used for indexing
+            String engineName = field.attributes().getOrDefault(KNNConstants.KNN_ENGINE, KNNEngine.DEFAULT.getName());
             KNNEngine knnEngine = KNNEngine.getEngine(engineName);
 
-            if (KNNEngine.NMSLIB.equals(knnEngine)) {
-                parameters.put(KNNConstants.SPACE_TYPE, fieldAttributes.getOrDefault(KNNConstants.SPACE_TYPE,
-                        SpaceType.DEFAULT.getValue()));
-
-                String efConstruction = fieldAttributes.get(KNNConstants.HNSW_ALGO_EF_CONSTRUCTION);
-                if (efConstruction != null) {
-                    parameters.put(KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION, Integer.parseInt(efConstruction));
-                }
-
-                String m = fieldAttributes.get(KNNConstants.HNSW_ALGO_M);
-                if (m != null) {
-                    parameters.put(KNNConstants.METHOD_PARAMETER_M, Integer.parseInt(m));
-                }
-
-            } else {
-                String parametersString = fieldAttributes.get(KNNConstants.PARAMETERS);
-                parameters.putAll(
-                        XContentFactory.xContent(XContentType.JSON).createParser(NamedXContentRegistry.EMPTY,
-                                DeprecationHandler.THROW_UNSUPPORTED_OPERATION, parametersString).map()
-                );
-            }
-
-
-            // Make Engine Name Into FileName
-            BinaryDocValues values = valuesProducer.getBinary(field);
+            // Create path where engine file will be stored
             String engineFileName = buildEngineFileName(state.segmentInfo.name, knnEngine.getLatestBuildVersion(),
```

Comment:
shall we move tmpEngineFileName close to where it is used?

Reply:
Yes, we can move.